Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -178,16 +178,19 @@ void FunctionDefinitionWriter::visit(AST::Structure& structDecl)
     m_structRole = std::nullopt;
 }
 
-void FunctionDefinitionWriter::visit(AST::Variable& variableDecl)
+void FunctionDefinitionWriter::visit(AST::Variable& variable)
 {
-    ASSERT(variableDecl.maybeTypeName());
-
-    m_stringBuilder.append(m_indent);
-    visit(*variableDecl.maybeTypeName());
-    m_stringBuilder.append(" ", variableDecl.name());
-    if (variableDecl.maybeInitializer()) {
+    if (variable.maybeTypeName())
+        visit(*variable.maybeTypeName());
+    else {
```

Comment:
Surely a variable declaration may have both a type name and an initializer?

Reply:
Yes, and that's correctly handled, no? We only branch here to choose where to get the type from, but we serialize the initializer below (regardless of whether or not we have a type)